internal/jujuapi/http_proxy.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2024 Canonical.
+package jujuapi
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/juju/names/v5"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/rpc"
+)
+
+type httpProxier struct {
+	jimm JIMM // interface
+}
+
+func (s *httpProxier) Authenticate(ctx context.Context, w http.ResponseWriter, req *http.Request) error {
+	// extract auth token
+	_, password, ok := req.BasicAuth()
+	if !ok {
+		return errors.E(errors.CodeUnauthorized, "authentication missing")
+	}
+	user, err := s.jimm.LoginWithSessionToken(ctx, password)
+	if err != nil {
+		return errors.E(errors.CodeUnauthorized, err)
+	}
+	// extract model uuid anche check permission
+	sPath, _ := strings.CutPrefix(req.URL.EscapedPath(), "/model")
```

Comment:
We wouldn't need to do this if we strip the /model before we reach the `httpProxier`.